src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs

```diff
@@ -40,12 +41,13 @@ public class DatadogHttpClient : IDatadogClient
         /// </summary>
         private static readonly UTF8Encoding UTF8 = new UTF8Encoding();
 
-        public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, string apiKey)
+        public DatadogHttpClient(DatadogConfiguration config, LogFormatter formatter, string apiKey, Action<Exception> exceptionHandler)
```

Comment:
The signature seems to suggest that the callback could expect all types of exceptions, while actually the only thing would be raised is CannotSendLogEventException. 

This can be confusing as user may expect this to be a generic exception handler which is not (e.g., if there is any exception on serializing the log this will not be called). The purpose of callback is to notify user that the logs were not sent. Can it be something more specific like Action<IEnumerable<LogEvent>> or if you want to convey additional information Action<CannotSendLogEventException>?

Reply:
Thanks to your comment, I noticed a log is skipped when its size is greater than `256 * 1024`:  https://github.com/DataDog/serilog-sinks-datadog-logs/blob/master/src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogHttpClient.cs#L72.

I have updated the code to handle all the exceptions raised by `DatadogHttpClient`. You can be notified when there is a network issue `CannotSendLogEventException` or when the log is skipped `TooBigLogEventException`. If the code is updated in the futur, I can add another kind of error without adding a new handler.

Reply:
Looks good to me. As I mentioned, the only concern I have is the signature of the action, which is to generic (any Exception). Maybe a base DatadogSinkException. But it's just my personal opinion.

As long as the documentation is clear so user of the callback knows what exception he could expect that is fine.